appservice/src/deploy/deployZip.ts

```diff
@@ -18,6 +18,10 @@ import { formatDeployLog } from './formatDeployLog';
 import { waitForDeploymentToComplete } from './waitForDeploymentToComplete';
 
 export async function deployZip(client: SiteClient, fsPath: string, configurationSectionName: string, aspPromise: Promise<AppServicePlan | undefined>): Promise<void> {
+    if (!(await fse.pathExists(fsPath))) {
+        throw new Error(localize('pathNotExist', 'Failed to deploy path that does not exist: {0}', fsPath));
+    }
+
```

Comment:
I don't get why throwing the error is being moved to another function.

Reply:
deployZip calls getZipFileToDeploy a few lines down. If i move the error, then it will also be thrown for the case when `if (FileUtilities.getFileExtension(fsPath) === 'zip') {`